agi-bin/inboundlookup.php

```diff
@@ -50,88 +51,122 @@ function inboundlookup_error($text,$tag='ERROR') {
 $number =  $argv[1];
 $name = '';
 $company = '';
+$apiFlag = false;
+$apiDir = '/usr/src/nethvoice/lookup.d';
+$apiFile = '';
 
 if (strlen($number)> 4) {
-    //get database data
-    $results = $db->getAll("SELECT * FROM inboundlookup","getRow",DB_FETCHMODE_ASSOC);
-    if (DB::isError($results) || empty($results)) {
-        inboundlookup_error ('Error getting inboundlookup data');
-        exit(1);
+    //search into dir custom API script(s)
+    if (is_dir($apiDir)) {
+        foreach (scandir($apiDir) as $file) {
+            //check if it's a file also it's executable
+            $apiFile = $apiDir . '/' . $file;
+            if (is_file($apiFile)) {
+                if (is_executable($apiFile)) {
+                    //call the exe file and check if it returns any valid json
+                    $shellCmd = escapeshellcmd($apiFile . ' ' . $number);
+                    $apiResult = shell_exec($shellCmd);
+                    //expects a json obj
+                    if ($apiResult != '') {
+                        $jsonData = json_decode($apiResult);
+                        $name = $jsonData->name;
+                        $company = $jsonData->company;
+                        if ($name == '' && $company == '') {
+                            $apiFlag = false;
```

Comment:
```suggestion
```
This is useless since it is already false and became true only before exit the cycle